delphi-frontend/src/main/java/au/com/integradev/delphi/cfg/block/ProtoBlockFactory.java

```diff
@@ -0,0 +1,395 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2025 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.cfg.block;
+
+import au.com.integradev.delphi.cfg.api.Block;
+import au.com.integradev.delphi.cfg.api.Branch;
+import au.com.integradev.delphi.cfg.api.Cases;
+import au.com.integradev.delphi.cfg.api.Finally;
+import au.com.integradev.delphi.cfg.api.Halt;
+import au.com.integradev.delphi.cfg.api.Linear;
+import au.com.integradev.delphi.cfg.api.Terminus;
+import au.com.integradev.delphi.cfg.api.UnconditionalJump;
+import au.com.integradev.delphi.cfg.api.UnknownException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.plugins.communitydelphi.api.ast.DelphiNode;
+
+public final class ProtoBlockFactory {
+  private ProtoBlockFactory() {
+    // Utility class
+  }
+
+  public static ProtoBlock exitBlock() {
+    return new ProtoBlock(TerminusImpl::new, (blocks, block) -> {});
+  }
+
+  public static ProtoBlock halt(DelphiNode terminator) {
+    return new ProtoBlock(HaltImpl::new, (blocks, block) -> ((HaltImpl) block).setData(terminator));
+  }
+
+  public static ProtoBlock withBranch(
+      DelphiNode terminator, ProtoBlock trueBlock, ProtoBlock falseBlock) {
+    return new ProtoBlock(
+        BranchImpl::new,
+        (blocks, block) ->
+            ((BranchImpl) block)
+                .setData(terminator, blocks.get(trueBlock), blocks.get(falseBlock)));
+  }
+
+  public static ProtoBlock finallyBlock(ProtoBlock successor, ProtoBlock finallySuccessor) {
+    return new ProtoBlock(
+        FinallyImpl::new,
+        (blocks, block) ->
+            ((FinallyImpl) block).setData(blocks.get(successor), blocks.get(finallySuccessor)));
+  }
+
+  public static ProtoBlock succeedingTo(ProtoBlock successor) {
```

Comment:
`succeedingTo` is a little confusing at first glance. I think the most important thing about the block this function creates is that it's totally linear. Maybe `linear`? Or even `before`?

```java
ProtoBlockFactory.succeedingTo(successor)
// vs
ProtoBlockFactory.linear(successor)
```